safe-global/web-core
# fix: decimal validation + high/low amount formatting

## What it solves

Decimal validation and amount formatting adjustments.

## How this PR fixes it

Decimal length is now validated according to the specified token decimals in the `<SendAssetsForm>` and `<SpendingLimitForm>`.

The logic for removing trailing 0s from formatted amounts in `safeFormatAmount` has been tweaked as the previous logic would return exponentials, breaking forms.

## How to test it

1. Attempt to create a transaction/add a spending limit with either `1.` with a large amount of decimal places. Observe the validation of decimal place length.
2. Attempt to send an asset of < 0.000000001. When clicking "Max", it should no longer display as an exponential.
